subgraphs/_reference_/src/sdk/protocols/bridge/pool.ts

```diff
@@ -0,0 +1,638 @@
+    }
+    return this.pool.inputToken != token.token!;
+  }
+
+  /**
+   * Recalculates the total value locked for this pool based on its current input token balance.
+   * This function will also update the protocol's total value locked based on the change in this pool's.
+   */
+  refreshTotalValueLocked(): void {
+    const tvl = this.getInputTokenAmountPrice(this.pool.inputTokenBalance);
+    this.setTotalValueLocked(tvl);
+  }
+
+  /**
+   * Updates the total value locked for this pool to the given value.
+   * Will also update the protocol's total value locked based on the change in this pool's.
+   */
+  setTotalValueLocked(newTVL: BigDecimal): void {
+    const delta = newTVL.minus(this.pool.totalValueLockedUSD);
+    this.addTotalValueLocked(delta);
+    this.save();
+  }
+
+  /**
+   * Adds the given delta to the total value locked for this pool.
+   * Will also update the protocol's total value locked based on the change in this pool's.
+   *
+   * @param delta The change in total value locked for this pool.
+   */
+  addTotalValueLocked(delta: BigDecimal): void {
+    this.pool.totalValueLockedUSD = this.pool.totalValueLockedUSD.plus(delta);
+    this.protocol.addTotalValueLocked(delta);
+    this.save();
+  }
+
+  /**
+   * Utility function to convert some amount of input token to USD.
+   *
+   * @param amount the amount of inputToken to convert to USD
+   * @returns The converted amount.
+   */
+  getInputTokenAmountPrice(amount: BigInt): BigDecimal {
+    const token = this.getInputToken();
+    const price = this.protocol.getTokenPricer().getTokenPrice(token);
+    token.lastPriceUSD = price;
+    token.save();
+
+    return amount.divDecimal(exponentToBigDecimal(token.decimals)).times(price);
+  }
+
+  /**
+   * Adds the given amount to the pool's input token balance. It will optionally
+   * update the pool's and protocol's total value locked. If not stated, will default to true.
+   *
+   * @param amount amount to be added to the pool's input token balance.
+   * @param updateMetrics optional parameter to indicate whether to update the pool's and protocol's total value locked.
+   */
+  addInputTokenBalance(amount: BigInt, updateMetrics: boolean = true): void {
+    const newBalance = this.pool.inputTokenBalance.plus(amount);
+    this.setInputTokenBalance(newBalance, updateMetrics);
+  }
+
+  /**
+   * Sets the pool's input token balance to the given amount. It will optionally
+   * update the pool's and protocol's total value locked. If not stated, will default to true.
+   *
+   * @param amount amount to be set as the pool's input token balance.
+   * @param updateMetrics optional parameter to indicate whether to update the pool's and protocol's total value locked.
+   */
+  setInputTokenBalance(
+    newBalance: BigInt,
+    updateMetrics: boolean = true
+  ): void {
+    this.pool.inputTokenBalance = newBalance;
+    if (updateMetrics) {
+      this.refreshTotalValueLocked();
+    }
+  }
+
+  getBytesID(): Bytes {
```

Comment:
It was because I initially made `Pool` and `PoolManager` to depend on an interface instead of of `Protocol` directly. And AssemblyScript doesn’t support defining fields in interfaces, only functions. So I had to make a getter instead.

But I think it is not necessary anymore, so I can remove it

Reply:
Oh, it’s actually in use by `account.ts`. I think I prefer to keep it, so it is `this.pool.GetBytesID()` instead of `this.pool.pool.id`